From 200381eeea85d3fef32d9863974b83cf5ffbdf88 Mon Sep 17 00:00:00 2001
From: Pau Ruiz Safont <pau.safont@citrix.com>
Date: Fri, 16 Sep 2022 14:48:09 +0100
Subject: [PATCH] xenopsd tests: don't run cancel_utils_test as unit test

In practice the watches code was not being exercised by the test in any
situation, so transforming it into an executable that doesn't get
compiled by default. It is still checked by the CI

Signed-off-by: Pau Ruiz Safont <pau.safont@citrix.com>
---
 ocaml/xenopsd/xc/cancel_utils_test.ml | 31 ++++++++++++---------------
 ocaml/xenopsd/xc/dune                 |  4 +---
 2 files changed, 15 insertions(+), 20 deletions(-)

diff --git a/ocaml/xenopsd/xc/cancel_utils_test.ml b/ocaml/xenopsd/xc/cancel_utils_test.ml
index a7f86a3447..459c0caa08 100644
--- a/ocaml/xenopsd/xc/cancel_utils_test.ml
+++ b/ocaml/xenopsd/xc/cancel_utils_test.ml
@@ -11,7 +11,10 @@
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Lesser General Public License for more details.
  *)
-open OUnit
+
+(* Tests whether the xenstore watches get cancelled successfully after the
+   timeout *)
+
 open Xenops_interface
 open Cancel_utils
 open Xenops_task
@@ -20,7 +23,7 @@ exception Did_not_cancel
 
 let tasks = Xenops_task.empty ()
 
-let xenstore_test xs _ =
+let xenstore_test xs =
   let task = Xenops_task.add tasks "test" (fun _ -> None) in
   let (_ : Thread.t) =
     Thread.create
@@ -34,21 +37,15 @@ let xenstore_test xs _ =
     let (_ : bool) =
       cancellable_watch (TestPath "/test/cancel") [] [] task ~xs ~timeout:3. ()
     in
+    Printf.printf "%s: failure: watch was not cancelled!" __MODULE__ ;
     raise Did_not_cancel
-  with Xenopsd_error (Cancelled _) -> (* success *)
-                                      ()
+  with Xenopsd_error (Cancelled _) ->
+    Printf.printf "%s: success: watch cancelled successfully" __MODULE__
 
-let _ =
-  let verbose = ref false in
-  Arg.parse
-    [("-verbose", Arg.Unit (fun _ -> verbose := true), "Run in verbose mode")]
-    (fun x -> Printf.fprintf stderr "Ignoring argument: %s\n" x)
-    "Test cancellation functions" ;
-  try
-    Xenstore.with_xs (fun xs ->
-        let suite = "cancel test" >::: ["xenstore" >:: xenstore_test xs] in
-        run_test_tt ~verbose:!verbose suite |> ignore
-    )
+let () =
+  try Xenstore.with_xs xenstore_test
   with Xs_transport.Could_not_find_xenstore ->
-    (* ignore test, we're not running on domain 0 *)
-    ()
+    Printf.printf
+      "%s: Xenstore not found, cannot test cancellable watches, are you \
+       running on dom0?"
+      __MODULE__
diff --git a/ocaml/xenopsd/xc/dune b/ocaml/xenopsd/xc/dune
index f7623d9837..1865b31f8d 100644
--- a/ocaml/xenopsd/xc/dune
+++ b/ocaml/xenopsd/xc/dune
@@ -114,14 +114,12 @@
  )
 )
 
-(test
+(executable
  (name cancel_utils_test)
- (package xapi-xenopsd-xc)
  (modules cancel_utils_test)
  (libraries
   cmdliner
   ezxenstore.core
-  ounit2
   threads.posix
   xapi-idl.xen.interface
   xapi-xenopsd
